api/v1beta2/awsmachine_types.go

```diff
@@ -197,6 +197,39 @@ type Ignition struct {
 	// +kubebuilder:default="2.3"
 	// +kubebuilder:validation:Enum="2.3";"3.0";"3.1";"3.2";"3.3";"3.4"
 	Version string `json:"version,omitempty"`
+
+	// IgnitionProxy defines proxy settings for Ignition.
+	// Only valid for Ignition versions 3.4 and above.
+	// +optional
+	Proxy *IgnitionProxy `json:"proxy,omitempty"`
+
+	// TLS defines TLS settings for Ignition.
+	// Only valid for Ignition versions 3.4 and above.
+	// +optional
+	TLS *IgnitionTLS `json:"tls,omitempty"`
+}
+
+// IgnitionTLS defines TLS settings for Ignition.
+type IgnitionTLS struct {
+	// CASources defines the list of certificate authorities to use for Ignition.
+	// THe value is the certificate bundle (in PEM format). The bundle can contain multiple concatenated certificates.
+	// Supported schemes are http, https, tftp, s3, arn, gs, and `data` (RFC 2397) URL scheme.
+	//
+	// +optional
+	CASources []string `json:"certificateAuthorities,omitempty"`
```

Comment:
Have you considered adding a maximal length to the slices/strings you're adding in this new API? Prevents users adding excessive amounts of data and will allow us later to add CEL validation to validate the content rather than using webhooks

Reply:
I didn't want to take guesses, it doesn't seem the ignition config has any limits really; are these limits documented somewhere?

Reply:
Proxy is being used in the ignition stub, so user data size limits would be a constraint. Seems like users could really hit that limit... although it looks like user data is compressed. Neat!

Reply:
Set a maximum on the list itself, do we want to restrict each URL as well?

Reply:
Typically where a sensible limit can be set, we would do so. For something like this, the data url would require you to use a type alias
```
CASources []CASource ...

// +kubebuilder:validation:MaxLength:=65536
type CASource string
```

Reply:
done!